tests/commands/test_bump_recipe.py

```diff
@@ -0,0 +1,79 @@
+"""
+:Description: Tests the `bump-recipe` CLI
+"""
+
+import pytest
+from click.testing import CliRunner
+from pyfakefs.fake_filesystem import FakeFilesystem
+
+from conda_recipe_manager.commands import bump_recipe
+from conda_recipe_manager.commands.utils.types import ExitCode
+from conda_recipe_manager.parser.recipe_parser import RecipeParser
+from tests.file_loading import get_test_path, load_recipe
+from tests.smoke_testing import assert_cli_usage
+
+
+def test_usage() -> None:
+    """
+    Smoke test that ensures rendering of the help menu
+    """
+    assert_cli_usage(bump_recipe.bump_recipe)
+
+
+@pytest.mark.parametrize(
+    "recipe_file, bumped_recipe_file",
+    [
+        ("simple-recipe.yaml", "bump_recipe/build_num_1.yaml"),
+        ("bump_recipe/build_num_1.yaml", "bump_recipe/build_num_2.yaml"),
+        ("bump_recipe/build_num_42.yaml", "bump_recipe/build_num_43.yaml"),
+        ("bump_recipe/build_num_-1.yaml", "simple-recipe.yaml"),
+    ],
+)
+def test_bump_recipe_cli(fs: FakeFilesystem, recipe_file: str, bumped_recipe_file: str) -> None:
+    """
+    Test for the case when build number is successfully incremented by 1.
+    :param fs: `pyfakefs` Fixture used to replace the file system
+    """
+    runner = CliRunner()
+    fs.add_real_directory(get_test_path(), read_only=False)
```

Comment:
I could have sworn we needed to use the delayed-evalutation `request.getfixturevalue("fs")` for this, but I guess that might not be the case? I guess you aren't using any other fixtures that could invoke the filesystem before `pyfakefs` is setup for the test.

Reply:
I can't find a clear answer/example from the documentation. I found one blog post that indicated that putting the fixture (`fs` in our case) at the end of the list should work. I guess `pytest` could be looking at the `kwargs` dictionary and might be going off of names and not the argument positions?

It bothers me how imprecise/unclear the `pytest` docs and other posts are about the nuance here.

Reply:
Finally: at least some sort of confirmation of what I just guessed: https://stackoverflow.com/questions/69636793/where-is-the-order-of-parameters-naming-of-them-in-a-pytest-function-document#:~:text=Fixtures%20are%20always%20looked%20up%20by%20name%2C%20so%20the%20order%20they%20appear%20in%20the%20argument%20list%20should%20not%20matter%20in%20principle.%20This%20is%20also%20true%20for%20the%20arguments%20given%20in%20pytest.mark.parametrize%2C%20as%20observed.

So it would appear `pytest` does rely on names over argument position. I guess we are in the clear.